pkg/reconciler/managed/reconciler.go

```diff
@@ -478,6 +479,9 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 	ctx, cancel := context.WithTimeout(context.Background(), managedReconcileTimeout)
 	defer cancel()
 
+	externalCtx, cancel := context.WithTimeout(ctx, externalClientTimeout)
+	defer cancel()
```

Comment:
I wonder whether we should be setting this here, or letting the ExternalConnector and ExternalClient implementations create their own child contexts with appropriate timeouts. What do you think?

Reply:
Letting the developer configure its APIs timeout sounds reasonable. But it should somehow be in sync with our reconciler context because in case they chose 2 minutes and our reconciler timeout is 1 minute, it doesn't make much sense. In cases where the normal response is >1 minute, even though the developer declared that, reconciliation would fail. Since it's a bit harder to figure out the chosen timeout from the context, it makes more sense to me that they just specify a timeout if they need to.

We can possibly allow configuration of the timeout via an `option`, which also affects the reconciler timeout. The defaults would be in place, but if they choose 90 seconds for API timeout, we'd add 30 seconds to that and say total reconciler timeout is 120 seconds while context passed to external client calls have 90 seconds timeout. So, we always make sure we're able to make our own calls. Just updated the PR to show what I mean.